drivers/pinmux/pinmux_esp32.c

```diff
@@ -164,7 +164,12 @@ static int pinmux_initialize(struct device *device)
 	u32_t pin;
 
 	for (pin = 0; pin < ARRAY_SIZE(pin_mux_off); pin++) {
-		pinmux_set(NULL, pin, 0);
+		/* pin 6 to 11 are configured for SPI by the second stage bootloader.
+		 * We should not change them, otherwise we can't access the flash
+		 * anymore. */
```

Comment:
If the point here is to disallow software from ever touching those pins through this driver, you probably want to put that rule into pinmux_set(), no?  

Honestly, I'm not sure I see much value to this initialization loop anyway.  If we have code enabled that cares about particular pin state, it will set it in its own initialization.  If not, it's probably better to rely on whatever the bootloader set for us.  Forcing "function zero" to be the default for everyone always is just going to cause collisions like this.  Maybe just remove the loop?

Reply:
I updated the comment. The idea is to keep the ability to reconfigure these pins during run-time.